Sources/SwiftProtobufCore/SwiftProtobufContiguousBytes.swift

```diff
@@ -18,23 +18,34 @@ import Foundation
 /// By conforming your own types to this protocol, you will be able to pass instances of said types
 /// directly to `SwiftProtobuf.Message`'s deserialisation methods
 /// (i.e. `init(serializedBytes:)` for binary format and `init(jsonUTF8Bytes:)` for JSON).
-// TODO: extend doc to include how this will be used in serialization once that API change has been finalised.
 public protocol SwiftProtobufContiguousBytes {
+    /// An initializer for a bag of bytes type.
+    ///
+    /// - Parameters:
+    ///   - repeating: the byte value to be repeated.
+    ///   - count: the number of times to repeat the byte value.
+    init(repeating: UInt8, count: Int)
+
+    /// The number of bytes in the bag of bytes.
+    var count: Int { get }
+
     /// Calls the given closure with the contents of underlying storage.
     ///
     /// - note: Calling `withUnsafeBytes` multiple times does not guarantee that
     ///         the same buffer pointer will be passed in every time.
     /// - warning: The buffer argument to the body should not be stored or used
     ///            outside of the lifetime of the call to the closure.
     func withUnsafeBytes<R>(_ body: (UnsafeRawBufferPointer) throws -> R) rethrows -> R
+
+    /// Calls the given closure with the contents of underlying storage.
+    ///
+    /// - note: Calling `withUnsafeBytes` multiple times does not guarantee that
+    ///         the same buffer pointer will be passed in every time.
+    /// - warning: The buffer argument to the body should not be stored or used
+    ///            outside of the lifetime of the call to the closure.
+    mutating func withUnsafeMutableBytes<R>(_ body: (UnsafeMutableRawBufferPointer) throws -> R) rethrows -> R
```

Comment:
The "initializing view" I mentioned above would be functionally equivalent to the following (but with compiler-enforced safety guarantees that we're just starting to plan out).
```
// Allocate `size` bytes after the current data and call the specified closure to initialize those bytes
mutating func withUnsafeAppendedBytes<R>(size: Int, initializer: (UnsafeMutableRawBufferPointer) throws -> R) rethrows -> R
```